app/main.py

```diff
@@ -1,3 +1,16 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Any, Callable
+
+
+def cache(func: Callable) -> Callable:
+    stock = {}
+
+    def wrapper(*args: Any) -> Any:
+        if args in stock:
+            print("Getting from cache")
+            return stock[args]
+        else:
+            print("Calculating new result")
+            result = func(*args)
+            stock[args] = result
+            return result
```

Comment:
No need to duplicate code inside if and else move it after if. See your checklist

Reply:
Gives an error message
![image](https://user-images.githubusercontent.com/126685656/226866743-1aacaac5-87c6-4fb9-882c-3eaac7c367e2.png)